rotkehlchen/chain/substrate/typing.py

```diff
@@ -0,0 +1,108 @@
+from enum import Enum
+from typing import Any, Dict, List, NamedTuple, NewType, Tuple, Type, Union
+
+from rotkehlchen.assets.asset import Asset
+from rotkehlchen.fval import FVal
+
+ChainId = NewType('ChainId', str)
+BlockNumber = NewType('BlockNumber', int)
+
+
+class KusamaNodeName(Enum):
+    """
+    TODO: this requires to be extended with OWN_NODE.
+    """
+    PARITY = 0
+    WEB3_FOUNDATION = 1
+
+    def __str__(self) -> str:
+        if self == KusamaNodeName.PARITY:
+            return 'parity'
+        if self == KusamaNodeName.WEB3_FOUNDATION:
+            return 'web3 foundation'
+        raise AssertionError(f'Unexpected KusamaNodeName: {self}')
+
+    def endpoint(self) -> str:
+        if self == KusamaNodeName.PARITY:
+            return 'wss://kusama-rpc.polkadot.io/'
+        if self == KusamaNodeName.WEB3_FOUNDATION:
+            return 'wss://cc3-5.kusama.network/'
+        raise AssertionError(f'Unexpected KusamaNodeName: {self}')
+
+    def is_open(self) -> bool:
+        if self == KusamaNodeName.PARITY:
+            return True
+        if self == KusamaNodeName.WEB3_FOUNDATION:
+            return True
+        raise AssertionError(f'Unexpected KusamaNodeName: {self}')
+
+
+NodeName = Union[KusamaNodeName]
+
+
+class ChainProperties(NamedTuple):
+    """These properties are populated straight from the blockchain
+    """
+    ss58_format: int
+    token: Asset  # from 'tokenSymbol'
+    token_decimals: FVal
+
+
+class SubstrateInterfaceAttributes(NamedTuple):
+    type_registry_preset: str
```

Comment:
Are there other attributes? If not is is not an overkill to make a named tuple for one attribute?

Reply:
Yes, there are [lots attributes](https://github.com/polkascan/py-substrate-interface/blob/master/substrateinterface/base.py#L376) and this is why I've implemented this namedtuple instead of just a `SubstrateChain.type_registry_preset`.

Reply:
ok!